src/RoomsSDKAdapter.js

```diff
@@ -36,42 +37,67 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
     };
   }
 
+  /**
+   * Create an observable that subscribes to room change events emitted via websocket and
+   * emits the changed room object.
+   *
+   * @private
+   * @param {string} ID ID of room to get
+   * @returns {Observable.<Room>} Observable that emits changes to the room
+   * @memberof RoomsSDKAdapter
+   */
+  addRoomChangeListener(ID) {
+    const roomChange$ = fromEvent(this.datasource.rooms, ROOM_UPDATED_EVENT);
+
+    return roomChange$.pipe(
+      // Is the room change event for our subscribed room?
+      filter((event) => event.data.id === ID),
+      // Event data doesn't have the room data in it, so we need to fetch manually
+      mergeMap(async () => this.fetchRoom(ID))
+    );
+  }
+
+  /**
+   * Builds the observable flow for fetching room data, then listening for
+   * websocket events about room changes.
+   *
+   * @private
+   * @param {string} ID
+   * @returns {Observable.<Room>}
+   * @memberof RoomsSDKAdapter
+   */
+  getRoomObservable(ID) {
+    const room$ = from(this.fetchRoom(ID));
+
+    return concat(
+      // Fetch Our Room Data
+      room$,
+      this.addRoomChangeListener(ID)
+    );
+  }
+
   /**
    * Returns an observable that emits room data of the given ID.
    *
+   * @public
    * @param {string} ID ID of room to get
    * @returns {Observable.<Room>}
    * @memberof RoomsSDKAdapter
    */
   getRoom(ID) {
     if (!(ID in this.getRoomObservables)) {
-      const source = Observable.create((observer) => {
-        // Start listening for room changes
-        this.datasource.rooms.listen();
-        this.datasource.rooms.on(ROOM_UPDATED_EVENT, () => {
-          // Room has updates, fetch and send
-          this.fetchRoom(ID)
-            .then((room) => observer.next(room))
-            .catch((error) => observer.error(error));
-        });
+      // Tell the sdk to start listening to room changes
+      this.datasource.rooms.listen();
```

Comment:
Shouldn't this be done at the adapter level?

Reply:
What do you mean? This is the rooms adapter.